pkg/resources/ray/ray.go

```diff
@@ -0,0 +1,56 @@
+package ray
+
+import (
+	"fmt"
+
+	dcv1alpha1 "github.com/dominodatalab/distributed-compute-operator/api/v1alpha1"
+	"github.com/dominodatalab/distributed-compute-operator/pkg/resources"
+)
+
+// Component is used to drive Kubernetes object generation for different ray types.
+type Component string
+
+const (
+	// ComponentNone indicates a generic ray resource.
+	ComponentNone Component = "none"
+	// ComponentHead indicates a ray head resource.
+	ComponentHead Component = "head"
+	// ComponentWorker indicates a ray worker resource.
+	ComponentWorker Component = "worker"
+	// ApplicationName defines the static name used to generate ray object metadata.
+	ApplicationName = "ray"
+)
+
+// InstanceObjectName returns the name that will be used to create most owned cluster resources.
+func InstanceObjectName(instance string, comp Component) string {
+	if comp == ComponentNone {
+		return fmt.Sprintf("%s-%s", instance, ApplicationName)
+	}
+
+	return fmt.Sprintf("%s-%s-%s", instance, ApplicationName, comp)
```

Comment:
Do issues like a name being too long bubble up to the CRD or something?

Reply:
All object names are limited to 253 characters, include CRs. If a name is too long, the K8s API will throw an error.

e.g. using service accounts

```console
$ kubectl create sa $(ruby -e "puts 'a' * 253")
serviceaccount/aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa created
$ kubectl create sa $(ruby -e "puts 'a' * 254")
The ServiceAccount "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa" is invalid: metadata.name: Invalid value: "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa": must be no more than 253 characters
```